pact/provider/core/src/main/java/org/arquillian/algeron/pact/provider/core/PactsReader.java

```diff
@@ -114,12 +123,51 @@ protected ContractsRetriever getContractsSource(final TestClass testClass) {
                     return pactLoaderClass.newInstance();
                 }
             } else {
-                final Annotation annotation = pactLoaders.iterator().next();
-                return annotation.annotationType().getAnnotation(ContractsSource.class).value()
-                        .getConstructor(annotation.annotationType()).newInstance(annotation);
+                if (isAnnotationPresent(pactLoaders)) {
+                    final Annotation annotation = pactLoaders.iterator().next();
+                    return annotation.annotationType().getAnnotation(ContractsSource.class).value()
+                            .getConstructor(annotation.annotationType()).newInstance(annotation);
+                } else {
+                    return getContractsRetriever(algeronProviderConfiguration);
+                }
             }
         } catch (final InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
             throw new IllegalStateException("Error while creating contracts source", e);
         }
     }
+
+    private boolean isAnnotationPresent(List<Annotation> pactLoaders) {
+        return pactLoaders.size() > 0;
+    }
+
+    private ContractsRetriever getContractsRetriever(AlgeronProviderConfiguration algeronProviderConfiguration) {
+        if (algeronProviderConfiguration != null && algeronProviderConfiguration.isRetrieverConfigurationSet()) {
+            final Map<String, Object> retrieverConfiguration = algeronProviderConfiguration.getRetrieverConfiguration();
+
+            if (retrieverConfiguration.containsKey(PROVIDER)) {
+                final ContractsRetriever contractRetriever = findContractRetriever((String) retrieverConfiguration.get(PROVIDER));
+                contractRetriever.configure(retrieverConfiguration);
+
+                return contractRetriever;
+            } else {
+                throw new IllegalArgumentException(String.format("Retrieving contracts are enabled, but configuration is not providing a %s property with provider name to be used.", PROVIDER));
+            }
+        }
+
+        throw new IllegalArgumentException("No retriever defined as annotation nor in arquillian configuration file.");
+    }
+
+    private ContractsRetriever findContractRetriever(String name) {
+
+        ServiceLoader<ContractsRetriever> serviceLoader = ServiceLoader.load(ContractsRetriever.class);
+
+        for (ContractsRetriever contractsRetriever : serviceLoader) {
```

Comment:
No, you cannot since it doesn't implements the standard `stream`. It just has an `spliterator` or `foreach`, but not `stream` method as `Collections` does.